bpf/conntrack/cleanup.go

```diff
@@ -79,7 +79,8 @@ func (l *LivenessScanner) Scan() {
 			// Look up the reverse entry, where we do the book-keeping.
 			revEntryBytes, err := l.ctMap.Get(ctVal.ReverseNATKey().AsBytes())
 			if err != nil && bpf.IsNotExists(err) {
-				// Forward entry exists but no reverse entry (and the grace period has expired).
+				// Forward entry exists but no reverse entry. We might have come across the reverse
```

Comment:
Think you need to check the creation timestamp here to make sure that the entry wasn't just created (or we might be racing with the BPF program creating the pair of keys).

Reply:
Imo not an issue - not sure if by desing or by accident ;-) but the original code did not deal with that - afaict in contrary to the comment - either. We always create the fwd entry _after_ we create the rev entry in BPF. So if we hit a fwd which does not have rev entry we must have deleted it or there was some weird inconsistency - in either case it seems correct to remove the orphaned fwd entry.

Otoh it is a bit brittle since it depends on how BPF code creates the entries and if that changes, this will break. I will at least update the comment.

However, Since we can/should not delete two keys in one iteration, I was pondering collecting all the rev keys that we asses as expired and defer deleting them at the end of the clean up. This way we would never remove rev key before fwd key.